rpmio/rpmlua.cc

```diff
@@ -1365,19 +1365,37 @@ static int luaopen_rpm(lua_State *L)
     return 1;
 }
 
-void check_deprecated(lua_State *L, const char *func, const char *deprecated_in)
+void check_deprecated(lua_State *L, const char *func,
+		      const char *deprecated_in, const char *removed_in)
 {
-    int warn = 1;
+    int err = 1;
+    const char *rpmversion = NULL;
     lua_getfield(L, LUA_REGISTRYINDEX, "RPM_PACKAGE_RPMVERSION");
-    if (lua_isstring(L, -1)) {
-	rpmver v1 = rpmverParse(lua_tostring(L, -1));
+    if (lua_isstring(L, -1))
+	rpmversion = lua_tostring(L, -1);
+    if (rpmversion) {
+	rpmver v1 = rpmverParse(rpmversion);
+	rpmver v2 = rpmverParse(removed_in);
+	if (v2 && rpmverCmp(v1, v2) < 0)
+	    err = 0;
+	rpmverFree(v2);
+	rpmverFree(v1);
+    }
+    lua_pop(L, 1);
+
+    if (err) {
+	luaL_error(L, "%s() is no longer available, use rpm.spawn() or rpm.execute() instead", func);
+    }
```

Comment:
The way this suppresses the warning from below is not quite clear to me and should probably be more obvious in the code.

Reply:
luaL_error() is a macro that causes the calling function to return immediately. I can certainly add a comment to make it clear.

Reply:
Yeah, then this whole code suddenly makes a lot more sense.

Reply:
Added a return with "not reached" comment there - this is commonly used as "return luaL_error();" but that wont work here because our function returns void. And that left me puzzling how does a macro know how to return? It doesn't - luaL_error() is not a macro contrary to what I said above, it exists via longjmp(). So a rather exotic bird.